src/main/java/com/terraformersmc/terrestria/feature/trees/ShrubFeature.java

```diff
@@ -0,0 +1,46 @@
+package com.terraformersmc.terrestria.feature.trees;
+
+import com.mojang.datafixers.Dynamic;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.IWorld;
+import net.minecraft.world.gen.chunk.ChunkGenerator;
+import net.minecraft.world.gen.chunk.ChunkGeneratorConfig;
+import net.minecraft.world.gen.feature.Feature;
+
+import java.util.Random;
+import java.util.function.Function;
+
+public class ShrubFeature extends Feature<ShrubFeatureConfig> {
+
+	public ShrubFeature(Function<Dynamic<?>, ? extends ShrubFeatureConfig> configDeserializer) {
+		super(configDeserializer);
+	}
+
+	@Override
+	public boolean generate(IWorld world, ChunkGenerator<? extends ChunkGeneratorConfig> generator, Random random, BlockPos pos, ShrubFeatureConfig config) {
+		//only generate on grass
+		if (world.getBlockState(pos.down()) != Blocks.GRASS_BLOCK.getDefaultState()) return false;
+
+		//height can be 1 or 2 blocks
+		int height = random.nextInt(2) + 1;
+		for (int y = 0; y < height; y++) {
+			world.setBlockState(pos.add(0, y, 0), config.woodState, 2);
+		}
```

Comment:
This would be better if it were just 2 setblock and a single if statement to eliminate the int instance and using a nextBollean